### PR TITLE
`Development`: Improve JPQL style for exercise repositories

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

I just fixed the style in one repository so I decided to go through some more repos to make the codebase more consistent and clean.

### Description

I enforced capitalisation (SELECT instead of select), simple variable declaration (:var instead of :#{#var}, explicit enum values, consistent newlines and indentations.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

No functional changes.

## Summary by CodeRabbit

- **Refactor**
	- Improved query formatting in `ExerciseGroupRepository` for better readability.
	- Enhanced code maintainability in `ExerciseRepository` by using enum constants and updating boolean logic.
	- Optimized database queries in `FileUploadSubmissionRepository` for more efficient data retrieval.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java [hunk 1]
```diff
@@ -335,10 +374,9 @@ AND EXISTS (
...
-            SELECT
-                CASE WHEN exerciseGroup IS NOT NULL
-                     THEN exerciseGroup.title
-                     ELSE exercise.title
+            SELECT CASE WHEN exerciseGroup IS NOT NULL
+                THEN exerciseGroup.title
+                ELSE exercise.title
                 END AS title
             FROM Exercise exercise
                 LEFT JOIN exercise.exerciseGroup exerciseGroup
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java:382]:
```suggestion
            SELECT CASE WHEN exerciseGroup IS NOT NULL
                THEN exerciseGroup.title
                ELSE exercise.title
            END AS title
            FROM Exercise exercise
                LEFT JOIN exercise.exerciseGroup exerciseGroup
```

**Review by Strohgelaender — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Review by coolchock — APPROVED** [review]
lgtm

**Review by MaximilianJG — APPROVED** [review]
code changes lgtm

File: src/main/java/de/tum/in/www1/artemis/repository/ExerciseGroupRepository.java [hunk 2]
```diff
@@ -23,7 +23,11 @@
 public interface ExerciseGroupRepository extends JpaRepository<ExerciseGroup, Long> {
 
     @EntityGraph(type = LOAD, attributePaths = { "exercises" })
-    @Query("SELECT e FROM ExerciseGroup e WHERE e.id = :exerciseGroupId")
+    @Query("""
+            SELECT e
+            FROM ExerciseGroup e
+            WHERE e.id = :exerciseGroupId
+            """)
     Optional<ExerciseGroup> findWithExercisesById(@Param("exerciseGroupId") Long exerciseGroupId);
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/repository/ExerciseGroupRepository.java:31]: Are you sure we even need the query here? This seems simple enough that spring can derive this.

File: src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java [hunk 3]
```diff
@@ -320,9 +357,11 @@ AND EXISTS (
...
      */
     @Query("""
-            SELECT e.id FROM Exercise e LEFT JOIN e.course c
+            SELECT e.id
+            FROM Exercise e
+                LEFT JOIN e.course c
             WHERE c.id = :courseId
-                """)
+            """)
     Set<Long> findAllIdsByCourseId(@Param("courseId") Long courseId);
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java:365]: Same here?

Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java:365]: Not really, we don't return the Exercise itself. We could use a projection, but that would require using a Wrapper Class for the ID afaik. As far as I'm concerned this is the best way to implement this.

**Review by Strohgelaender — APPROVED** [review]

**Review by nityanandaz — APPROVED** [review]
Code review: looks good

**Pull request merged**